dbt/adapters/athena/python_submissions.py

```diff
@@ -162,18 +144,48 @@ def submit(self, compiled_code: str) -> Any:
                     "Result"
                 ]
             except Exception as e:
-                LOGGER.error(f"Unable to poll execution status: Got: {e}")
+                LOGGER.error(f"Unable to retrieve results: Got: {e}")
                 result = {}
-        self.spark_connection.release_session_lock(self.session_id)
         return result
 
+    def poll_until_session_idle(self) -> None:
+        """
+        Polls the session status until it becomes idle or exceeds the timeout.
+
+        Raises:
+            DbtRuntimeError: If the session chosen is not available or if it does not become idle within the timeout.
+        """
+        polling_interval = self.polling_interval
+        while True:
+            timer = 0
+            session_status = self.get_current_session_status()["State"]
+            if session_status in ["TERMINATING", "TERMINATED", "DEGRADED", "FAILED"]:
+                LOGGER.debug(
+                    f"The session: {self.session_id} was not available. Got status: {session_status}. Will try with a different session."
+                )
+                self.spark_connection.remove_terminated_session(self.session_id)
+                if "session_id" in self.__dict__:
+                    del self.__dict__["session_id"]
+                break
+            if session_status == "IDLE":
+                break
+            time.sleep(polling_interval)
+            timer += polling_interval
+            if timer > self.timeout:
+                LOGGER.debug(
+                    f"Session {self.session_id} did not become free within {self.timeout} seconds. Will try with a different session."
+                )
+                if "session_id" in self.__dict__:
+                    del self.__dict__["session_id"]
+                break
+
     def poll_until_execution_completion(self, calculation_execution_id: str) -> Any:
         """
-        Poll the status of a calculation execution until it is completed, failed, or cancelled.
+        Poll the status of a calculation execution until it is completed, failed, or canceled.
```

Comment:
praise: Good catch on the spelling for canceled! Just verified it here https://docs.aws.amazon.com/athena/latest/APIReference/API_CalculationStatus.html